### PR TITLE
chore(restify): update restify version to 11.1.0 and re-enable unit test in node@18

### PR DESCRIPTION
**Pull request by chigia001**
## Which problem is this PR solving?

- restify already release new version 11.1.0 and since 10.0.0 they already add support for node@18
- instrumentation-restify still testing with version restify@8.6.1, which don't suppport node@18

## Short description of the changes

- verify unit test still working with [restify@9.1.0](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1576/commits/ef7c9c20c73f3654738ca80de7f001a9ec5c2906) on node@14
- verify unit test still working with [restify@10.0.0](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1576/commits/f8c327f6ba94f7064dc0ad6f47b3230401f3559b) and re-enable unit test for node@18
- verify unit test still working with [restify@11.1.0](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1576/commits/bab09d4e02032c29e594220f088a334bf7f7eacf)
- update SUPPORTED_VERSIONS to `>=4.0.0 <12`

### CODE REVIEW

**Review by pichlermarc — APPROVED** [review]

File: .github/workflows/test-all-versions.yml [hunk 1]
```diff
@@ -22,8 +22,6 @@ jobs:
         include:
           - node: "18"
             lerna-extra-args: >-
-              --ignore @opentelemetry/instrumentation-fastify
```
Comment (blumamir) [inline .github/workflows/test-all-versions.yml:25]: This PR is for restify but it also remove ignoring fastify. Is it intentional?

File: plugins/node/opentelemetry-instrumentation-restify/src/constants.ts [hunk 2]
```diff
@@ -24,4 +24,4 @@ export const RESTIFY_METHODS = [
   'patch',
 ];
 export const MODULE_NAME = 'restify';
-export const SUPPORTED_VERSIONS = ['>=4.0.0 <9'];
+export const SUPPORTED_VERSIONS = ['>=4.0.0 <12'];
```
Comment (blumamir) [inline plugins/node/opentelemetry-instrumentation-restify/src/constants.ts:27]: This should also be updated in the [instrumentation package README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-restify#supported-versions)

File: plugins/node/opentelemetry-instrumentation-restify/.tav.yml [hunk 3]
```diff
@@ -1,3 +1,8 @@
 restify:
-  versions: "4.3.4 || 5.2.0 || 6.4.0 || 7.7.0 || ^8.4.0"
-  commands: npm run test
+  - versions: "^10.0.0 || ^11.1.0"
+    node: ">=18"
```
Comment (blumamir) [inline plugins/node/opentelemetry-instrumentation-restify/.tav.yml:3]: Should we still test versions 10 and 11 for older node versions? According to [restify package.json](https://github.com/restify/node-restify/blob/cbd16efa3be36e7888ecccc15ee28eaa8fa6c5ef/package.json#L93) it is supported:

```
...
  "engines": {
    "node": ">=10.0.0"
  },
...
```

File: plugins/node/opentelemetry-instrumentation-restify/package.json [hunk 4]
```diff
@@ -56,7 +56,7 @@
     "@types/semver": "^7.3.12",
     "mocha": "7.2.0",
     "nyc": "15.1.0",
-    "restify": "8.6.1",
+    "restify": "^11.1.0",
```
Comment (blumamir) [inline plugins/node/opentelemetry-instrumentation-restify/package.json:59]: I think renovate will create a followup PR to pin this dependency. Might be better to bump it to new version but keep it pinned

File: .github/workflows/test-all-versions.yml (hunk 1 above)
Comment (chigia001) [reply .github/workflows/test-all-versions.yml:25]: yes, I miss this part in my PR in https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1568
Do you want to split this?

File: plugins/node/opentelemetry-instrumentation-restify/.tav.yml (hunk 3 above)
Comment (chima-mckinsey) [reply plugins/node/opentelemetry-instrumentation-restify/.tav.yml:3]: update in the bellow segment for `node: "<18"`

File: .github/workflows/test-all-versions.yml (hunk 1 above)
Comment (blumamir) [reply .github/workflows/test-all-versions.yml:25]: Cool , no need to split

**Review by blumamir — APPROVED** [review]
LGTM
Thanks!

**Pull request merged**